Admin/Admin.php

```diff
@@ -923,11 +923,37 @@ public function buildDatagrid()
      * Returns the name of the parent related field, so the field can be use to set the default
      * value (ie the parent object) or to filter the object.
      *
-     * @return string the name of the parent related field
+     * @return null|string
+     *
+     * @throws \InvalidArgumentException
      */
     public function getParentAssociationMapping()
     {
-        return $this->parentAssociationMapping;
+        if (is_array($this->parentAssociationMapping) && $this->getParent()) {
+            $parent = $this->getParent()->getCode();
+            if (array_key_exists($parent, $this->parentAssociationMapping)) {
+                return $this->parentAssociationMapping[$parent];
+            } else {
+                throw new \InvalidArgumentException(sprintf(
+                    "There's no association between %s and %s.",
+                    $this->getCode(),
+                    $this->getParent()->getCode()
+                ));
+            }
+        } else {
+            return $this->parentAssociationMapping;
+        }
+    }
+
+    /**
+     * @param string $code
+     * @param string $value
+     *
+     * @return $this
+     */
+    public function addParentAssociationMapping($code, $value)
+    {
+        $this->parentAssociationMapping[$code] = $value;
```

Comment:
@robhunt3r, can you fix that quickly ? We are trying to merge all [these PRs](https://github.com/sonata-project/SonataAdminBundle/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+milestone%3A2.4+) before doing some big changes on the project.